src/diagnostics_component/utils/file_info_utils/src/file_info_utils.c

```diff
@@ -168,13 +168,11 @@ bool FileInfoUtils_FillFileInfoWithNewestFilesInDir(FileInfo* logFiles, size_t l
 
     if (!succeeded)
     {
+        // Free all the file names, if allocated.
```

Comment:
L158: there's MAX_FILES_TO_SCAN and there's logFileSize.  What happens MAX_FILES_TO_SCAN is greater (or much greater!) than logFileSize?